### ISSUE TITLE
max-len does not handle astral symbols correctly

### ISSUE THREAD

**Issue by lencioni** [eslint/eslint#2370]
In one of my files, I have the following comment:

```js
        // Astral symbols like e.g. '𝐀' (U+1D400 MATHEMATICAL BOLD CAPITAL A)
        // and '𝐁' (U+1D401 MATHEMATICAL BOLD CAPITAL B) take up two code points
        // in JavaScript strings (which are encoded in UCS-2). In UTF-8, which
        // we use on the server, they only take up one code point.
        //
        // More info:
        // https://mathiasbynens.be/notes/javascript-encoding
```

The max-len rule complains that the second line of this comment is longer than 80 characters, but it is not. This is happening because the 𝐁 takes up two code points in JS strings.

Here's a good presentation on the topic: http://2014.jsconf.eu/speakers/mathias-bynens-javascript-loves-unicode.html

In our codebase, we deal with these characters by using this punycode library: https://github.com/bestiejs/punycode.js/

**Comment by nzakas** [comment]
This is interesting. I'm not sure it's worth making a change to support this. Needing to convert each file would add some processing time, and I'm just not convinced that this case is worth that overhead.

When/if Node gets ES6 support we can use the `u` regex flag, but until then, I'm inclined to let this edge case stand.

**Comment by lencioni** [comment]
> When/if Node gets ES6 support we can use the u regex flag, but until then, I'm inclined to let this edge case stand.

Yes, this is exactly what I was thinking.

**Comment by lencioni** [comment]
In the meantime, it might be worth adding a note to the rule documentation.

**Comment by nzakas** [comment]
Agreed.
